libs/cardano-ledger-core/src/Cardano/Ledger/BaseTypes.hs

```diff
@@ -737,6 +739,14 @@ data Mismatch (r :: Relation) a = Mismatch
   }
   deriving (Eq, Ord, Show, Generic, NFData, ToJSON, FromJSON, NoThunks)
 
+swapMismatch :: Mismatch r a -> Mismatch r a
+swapMismatch m@(Mismatch _ _) =
+  let Mismatch {..} = m
+   in Mismatch
+        { mismatchSupplied = mismatchExpected
+        , mismatchExpected = mismatchSupplied
+        }
```

Comment:
Thank you! This makes it easier to understand what's going on.

Reply:
I agree 💯

Reply:
I tried this with specifying `{Enc,Dec}CBORGroup` instance for `(a, a)` and the types end up disagreeing because the `Coders` require the element to be of type `Mismatch`. Let me know if you have an idea around it. `<$>` and `MapE` also won't work.

Reply:
I have updated the implementation with your suggestion. I would like to self-review this PR once again just to be sure there are no mistakes before making it open to review again.